cargo-shuttle/src/lib.rs

```diff
@@ -1798,24 +1798,31 @@ impl Shuttle {
         Ok(())
     }
 
-    async fn project_create(&self, name: Option<String>) -> Result<()> {
-        let Some(ref name) = name else {
+    async fn project_create(&mut self, project_args: &ProjectArgs) -> Result<()> {
+        let Some(ref name) = project_args.name else {
             bail!("Provide a project name with '--name <name>'");
         };
 
+        self.ctx.load_local_internal_config(project_args)?;
+
         let client = self.client.as_ref().unwrap();
         let r = client.create_project(name).await?;
 
+        let raw_json = r.raw_json.clone();
+        let project = r.into_inner();
+
         match self.output_mode {
             OutputMode::Normal => {
-                let project = r.into_inner();
                 println!("Created project '{}' with id {}", project.name, project.id);
             }
             OutputMode::Json => {
-                println!("{}", r.raw_json);
+                println!("{}", raw_json);
             }
         }
 
+        self.ctx.set_project_id(project.id);
+        self.ctx.save_local_internal()?;
```

Comment:
If we are calling `project create` (which can be called from any dir) from a directory without a rust project, this would write a `.shuttle/config.toml` file in a location where it is not wanted. We might need a way to save the config only if it existed when trying to load it.